### ISSUE TITLE
Snap Camera doesn't load any lenses

### ISSUE THREAD

**Issue by Speedy2662** [ptrumpis/snap-camera-server#132]
**System (please complete the following information):**
 - OS: Windows 10
 - Server Version: 3.2.1
 - Snap Camera Version: 1.21
 - Signature Patch applied: yes

The Docker server is online;
![image](https://github.com/ptrumpis/snap-camera-server/assets/8947452/aa111096-01b7-4028-9cbf-6f8b459687d7)

What I see when I open Snap Camera:
![image](https://github.com/ptrumpis/snap-camera-server/assets/8947452/45ebc37f-9c63-4052-aeb4-9da52825de78)

[Log.txt](https://github.com/ptrumpis/snap-camera-server/files/14937188/Log.txt)

**Comment by ptrumpis** [comment]
Is https://studio-app.snapchat.com/ reachable in your browser?

Are you sure snap camera was fully closed?

```sh
 pkill -x "Snap Camera"
```

**Comment by Speedy2662** [comment]
> Is https://studio-app.snapchat.com/ reachable in your browser?
> 

I cannot access https://studio-app.snapchat.com/.
`An error occurred during a connection to studio-app.snapchat.com. PR_END_OF_FILE_ERROR`

> Are you sure snap camera was fully closed?

Yes, the process was closed, not just minimised. I didn't touch Snap Camera until the server was fully online.

**Comment by ptrumpis** [comment]
Looks like an SSL problem.

Did you use `AutoConfig.exe` or manual configuration?

**Comment by ptrumpis** [comment]
What about these links, are both working or is one of them failing?

http://studio-app.snapchat.com/
http://127.0.0.1/

**Comment by Speedy2662** [comment]
I used AutoConfig. I can also confirm I installed OpenSSL-3.2.1 prior to using AutoConfig

**Comment by Speedy2662** [comment]
> What about these links, are both working or is one of them failing?

http://127.0.0.1/ loads Snap Camera Server Files, the other studio-app link does not

**Comment by ptrumpis** [comment]
If the domain fails there might be a problem with the `hosts` file located in `%SYSTEMROOT%\System32\drivers\etc` 

The entry is missing or it's not getting honored.
Are you behind a proxy or vpn?

**Comment by Speedy2662** [comment]
I am not using a proxy, vpn, or any kind of antivirus or firewall software which doesn't come with Windows

**Comment by ptrumpis** [comment]
These commands may or may not help.

In Terminal
```
ipconfig /flushdns
```

In PowerShell
```
Clear-DnsClientCache
```

**Comment by Speedy2662** [comment]
I looked into the hosts file and as you suspected there was no mention of snapchat anywhere.

After adding
`127.0.0.1 studio-app.snapchat.com`
to the end of the file, executing the two commands you mention in your last comment and restarting Snapchat, it appears to work now

![image](https://github.com/ptrumpis/snap-camera-server/assets/8947452/f9e536a0-997c-44a3-b03a-b62f6f86207f)


I must note that editing the hosts file was for some reason way more difficult than I ever remember it being - running Notepad++ / Notepad as administrator was not enough to let me edit or save it.

I had to make a copy on my desktop, edit that then drag it back into the directory replacing the original. Odd!

**Comment by Speedy2662** [comment]
I have a quick question and I'm unsure if it might be related to the above - but are AI lenses such as 'Cubism Art by Snapchat' supposed to work? When I select it nothing happens, and I don't think I saw any mention regarding these :)

**Comment by ptrumpis** [comment]
Beauty Cartoon should work.
Cubism Art might work after additional patching with a tool.

You need to select the non working lens and attach your log file again, then I could tell you the exact problem.

**Comment by Speedy2662** [comment]
[Log.txt](https://github.com/ptrumpis/snap-camera-server/files/14937756/Log.txt)

Here it is. Appreciate your help and work on this project brother, thank you

**Comment by Speedy2662** [comment]
[Log.txt](https://github.com/ptrumpis/snap-camera-server/files/14937759/Log.txt)
Here's the log with Beauty Cartoon last

**Comment by ptrumpis** [comment]
> [Log.txt](https://github.com/ptrumpis/snap-camera-server/files/14937759/Log.txt) Here's the log with Beauty Cartoon last

You should be able to resolve this by following the instructions here:
https://github.com/ptrumpis/snap-camera-server/wiki/Updating-Snap-Camera-(Unofficial)

Or by simply downloading the `CoreResources.bundle`  Folder from the Asset Repository.

**Comment by ptrumpis** [comment]
> [Log.txt](https://github.com/ptrumpis/snap-camera-server/files/14937756/Log.txt)
> 
> Here it is. Appreciate your help and work on this project brother, thank you
> 
> Edit: I think it has to do with the blending mode as you answered in another thread. I'll attach logs for another which doesn't work

This is a little bit more complicated and will hopfully be resolved with the next v4.0 server version.
You would have to patch the lens as it contains an unknown blend mode.

Another user made a video tutorial about it:
https://github.com/ptrumpis/snap-camera-server/discussions/125

That user also provided some patched lenses:
https://github.com/ptrumpis/snap-camera-server/discussions/119

**Comment by Speedy2662** [comment]
> Or by simply downloading the `CoreResources.bundle` Folder from the Asset Repository.
I actually have done! I will attempt the update guide now, thanks.

I will probably wait for the 4.0 version to fix the blend modes - I downloaded the #119 lenses you linked but I haven't looked into how to import them yet. Thanks for providing the link there!

Once again, appreciate your help.
